extension-localization/src/main/java/com/mapbox/maps/extension/localization/Localization.kt

```diff
@@ -19,8 +19,13 @@ import java.util.*
  * `["format",["coalesce",["get","name_zh-Hans"],["get","name"]],{}]`
  */
 internal fun setMapLanguage(locale: Locale, style: StyleInterface, layerIds: List<String>?) {
+  var convertedLocale = "name_${locale.language}"
+  if (!isSupportedLanguage(convertedLocale)) {
+    Logger.e(TAG, "Locale: $locale is not supported.")
```

Comment:
would a simple logging message enough for notifying the local not supported error?

Reply:
How should user determine if locale is supported or not, do we have some public API for that?

Reply:
+1, maybe consider throwing IllegalArgumentException or some Unsupported* instead of ignoring the call completely

Reply:
@yunikkk @pengdev 

The issue is that the input could also be user driven. A developer could configure his map with:

```
Locale.getDefault()
```

And if the user has his device in a unsupported locale, it would crash the application :) which isn't ideal.
For this reason, we ignore silently and provide a fallback mechanism.

--------

This said, I do think we can provide a better experience for developers by exposing a listener callback. This way they can react to unsupported language configurations. Considering we will be rolling out new localization feature in Q1/Q2 and deprecating this, I think we should take this as a lessons learned for that new API but leave this as-is for now.